Project.toml

```diff
@@ -4,10 +4,12 @@ authors = ["Milan Kloewer"]
 version = "0.5.0"
 
 [deps]
+Checkpointing = "eb46d486-4f9c-4c3d-b445-a617f2a2f1ca"
 Dates = "ade2ca70-3891-5945-98fb-dc099432e06a"
 Interpolations = "a98d9a8b-a2ab-59e6-89dd-64a1c18fca59"
 NetCDF = "30363a11-5582-574a-97bb-aa9a979735b9"
 Parameters = "d96e819e-fc66-5662-9728-84c9c7592b0a"
+Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
```

Comment:
Both Checkpointing and Plots (and Enzyme) should probably be an extension? That way they are only a soft dependency and a general user does not depend on them but you can use/extend that functionality without a hard dependency on it, Oceananigans does that for Enzyme, SpeedyWeather does that for Makie

Reply:
I can remove Plots, but if I don't use Checkpointing I won't be able to run include(".src/ShallowWaters.jl"). I'll elaborate in a different comment